app/main/views/frameworks.py

```diff
@@ -363,3 +382,32 @@ def framework_updates_email_clarification_question(framework_slug):
 
     flash('message_sent', 'success')
     return framework_updates(framework['slug'])
+
+
+@main.route('/frameworks/<framework_slug>/agreement', methods=['GET'])
+@login_required
+def framework_agreement(framework_slug):
+    framework = data_api_client.get_framework(framework_slug)['frameworks']
+    if framework['status'] not in ['standstill', 'live']:
+        abort(404)
+
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    supplier_framework = data_api_client.get_supplier_framework_info(
+        current_user.supplier_id, framework_slug
+    )['frameworkInterest']
+    if not supplier_framework['onFramework']:
+        abort(404)
+
+    return render_template(
+        "frameworks/agreement.html",
+        framework=framework,
+        supplier_framework=supplier_framework,
+        **template_data
+    ), 200
+
+
+@main.route('/frameworks/<framework_slug>/agreement', methods=['POST'])
+@login_required
+def upload_framework_agreement(framework_slug):
+    pass
```

Comment:
This could return a `501: Not Implemented` response rather than break.

But it doesn't really matter as I know it's going to be filled in already anyway.